code/ui/InspectMenu.cs

```diff
@@ -21,33 +22,161 @@ public bool IsShowing
 
         private bool isShowing = false;
 
-        private Label InspectLabel { set; get; }
+        private Header header { set; get; }
+
+        private Content content { set; get; }
+
+        private Footer footer { set; get; }
+
+        private ConfirmationPanel confirmationPanel { get; set; }
 
         public InspectMenu()
         {
             Instance = this;
+            IsShowing = false;
+
             StyleSheet.Load("/ui/InspectMenu.scss");
 
-            InspectLabel = Add.Label("");
+            header = new Header(this);
+            header.AddClass("hide");
 
-            IsShowing = false;
+            content = new Content(this);
+            content.AddClass("hide");
+
+            footer = new Footer(this);
+            content.AddClass("hide");
+
+            confirmationPanel = new ConfirmationPanel(this);
         }
 
         public void InspectCorpse(TTTPlayer deadPlayer, bool isIdentified)
         {
             IsShowing = true;
 
-            // TODO: Setup proper hud, for now everything is just being thrown into "InspectLabel"
             if (isIdentified)
             {
-                InspectLabel.Text = $"{deadPlayer.GetClientOwner()?.Name}\n"
-                    + $"{deadPlayer.Role.Name}";
+                confirmationPanel.SetClass("hide", true);
+
+                header.SetPlayer(deadPlayer);
+                header.SetClass("hide", false);
+
+                content.SetPlayer(deadPlayer);
+                content.SetClass("hide", false);
+
+                footer.SetClass("hide", false);
```

Comment:
Is there a way we can setup this UI to be a single element that gets "hidden"/"unhidden"?

Reply:
What do you mean exactly? Currently, it's one UI element, that includes more or less 4 panels, 3 of them should just be shown if the corpse was confirmed. The other, if not confirmed. I think that can be handled better as well (just 2 panels), if that's what you meant?

Reply:
If the 3 panels could have one single "hide" call made on them. As in compile the 3 panels into a single "component", then just "hide' the single compontent.

Reply:
Done